feat(react-scheduler): allow appointment resizing (#1932)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/reference/drag-drop-provider.md

Name | Type | Default | Description
-----|------|---------|------------
allowDrag? | (appointmentData: [AppointmentModel](./scheduler.md#appointmentmodel)) => boolean | () => true | A function that specifies draggable appointments.
allowResize? | (appointmentData: [AppointmentModel](./scheduler.md#appointmentmodel)) => boolean | () => true | A function that specifies resizable appointments.
draftAppointmentComponent | ComponentType&lt;[DragDropProvider.DraftAppointmentProps](#dragdropproviderdraftappointmentprops)&gt; | | A component that renders the appointment being dragged.
sourceAppointmentComponent | ComponentType&lt;[DragDropProvider.SourceAppointmentProps](#dragdropprovidersourceappointmentprops)&gt; | | A component that renders a copy of the appointment being dragged in its previous location.

> resizeComponent | ComponentType&lt;[DragDropProvider.ResizeProps](#dragdropproviderresizeprops)&gt; | | A component that renders a handler of resize appointment.

Reviewer: ```suggestion
resizeComponent | ComponentType&lt;[DragDropProvider.ResizeProps](#dragdropproviderresizeprops)&gt; | | A component that renders a handle used to resize the appointment.
```

data | [AppointmentModel](./scheduler.md#appointmentmodel) | Specifies the appointment's data.
style | object | Configures the appointment's geometry and position.
type | string | Specifies the appointment's type.

> leftSlice | boolean | True if an appointment does not start on the current view.

Reviewer: ```suggestion
leftSlice | boolean | **true** if the appointment is continued from the previous day/week/month/year.
```

> rightSlice | boolean | True if an appointment does not end on the current view.

Reviewer: ```suggestion
rightSlice | boolean | **true** if the appointment continues on the next day/week/month/year.
```

Field | Type | Description
------|------|------------
data | [AppointmentModel](./scheduler.md#appointmentmodel) | Specifies the appointment's data.
type | string | Specifies the appointment's type.

### DragDropProvider.ResizeProps

> Describes properties of the component that renders a handler of resize appointment.

Reviewer: ```suggestion
Describes properties of the component that renders a handle used to resize the appointment.
```

Field | Type | Description
------|------|------------

> position | 'start' &#124; 'end' | Specifies a position for handler of resize.

Reviewer: ```suggestion
position | 'start' &#124; 'end' | Specifies the handle's position in the appointment.
```

> appointmentType | 'vertical' &#124; 'horizontal' | Specifies an appointment type.

Reviewer: ```suggestion
appointmentType | 'vertical' &#124; 'horizontal' | Specifies whether the appointment is vertical or horizontal.
```

-----|------------|------------
DragDropProvider.DraftAppointment | [DragDropProvider.DraftAppointmentProps](#dragdropproviderdraftappointmentprops) | A component that renders the appointment being dragged.
DragDropProvider.SourceAppointment | [DragDropProvider.SourceAppointmentProps](#dragdropprovidersourceappointmentprops) | A component that renders a copy of the appointment being dragged in its previous location.

> DragDropProvider.Resize | [DragDropProvider.ResizeProps](#dragdropproviderresizeprops) | A component that renders a handler of resize appointment.

Reviewer 2: DragDropProvider.ResizeComponent ?

Author: What do you mean?

Reviewer 2: Oh, it's the 'Plugin Components' section. Everything is OK in this case.

Reviewer 3: handler -> handle

> DragDropProvider.Resize | [DragDropProvider.ResizeProps](#dragdropproviderresizeprops) | A component that renders a handler of resize appointment.

Reviewer 3: handler -> handle

> DragDropProvider.Resize | [DragDropProvider.ResizeProps](#dragdropproviderresizeprops) | A component that renders a handle of an appointment being resized.

Reviewer: ```suggestion
DragDropProvider.Resize | [DragDropProvider.ResizeProps](#dragdropproviderresizeprops) | A component that renders the handle of the appointment being resized.
```
